source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1711 @@
+   G x = (fx-is-upbnd , fx-is-least-upbnd)
+    where
+     fx-is-upbnd : (f x is-an-upper-bound-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+     fx-is-upbnd (b , e) = ↓-to-fx-upbnd e
+      where
+       ↓-to-fx-upbnd : (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ x) holds) holds
+                     → (β b ≤ f x) holds
+       ↓-to-fx-upbnd = ∥∥-rec (holds-is-prop (β b ≤ f x)) ↓-to-fx-upbnd'
+        where
+         ↓-to-fx-upbnd' : Σ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ x) holds
+                        → (β b ≤ f x) holds
+         ↓-to-fx-upbnd' (a' , o , r) =
+          transitivity-of L (β b) (f a') (f x)
+                          (≤ᴮ-to-≤ (⌜ ≃-Lift 𝓤 (b ≤ᴮ f a') ⌝⁻¹ o))
+                          (f-mono a' x r)
+     fx-is-least-upbnd : ((u , _) : upper-bound (ϕ ↓ x , β ∘ ↓-to-base ϕ x))
+                       → (f x ≤ u) holds
+     fx-is-least-upbnd (u , is-upbnd) =
+      (is-least-upper-boundᴮ (f x))
+       (u , λ z → is-upbnd (⌜ ↓ᴮf-equiv-↓-tot x ⌝ z))
+   H : (x : ⟨ L ⟩) → (Γ ϕ i) x ＝ f x
+   H x =
+    reindexing-along-equiv-＝-sup
+     L (id , id-is-equiv (ϕ ↓ x)) (β ∘ ↓-to-base ϕ x)
+     ((Γ ϕ i) x) (f x) (sup-of-small-fam-is-lub  L (β ∘ ↓-to-base ϕ x) (i x))
+     (G x)
+
+ ind-def-from-monotone-map : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                           → is-monotone-endomap L f
+                           → 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)
+ ind-def-from-monotone-map f f-mono =
+  pr₁ (monotone-map-give-local-ind-def f f-mono)
+
+ local-from-monotone-map : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                         → (f-mono : is-monotone-endomap L f)
+                         → is-local (ind-def-from-monotone-map f f-mono)
+ local-from-monotone-map f f-mono =
+  pr₁ (pr₂ (monotone-map-give-local-ind-def f f-mono))
+
+ local-ind-def-is-section-of-Γ : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                               → (f-mono : is-monotone-endomap L f)
+                               → (x : ⟨ L ⟩)
+                               → (Γ (ind-def-from-monotone-map f f-mono)
+                                    (local-from-monotone-map f f-mono)) x
+                               ＝ f x
+ local-ind-def-is-section-of-Γ f f-mono =
+   pr₂ (pr₂ (monotone-map-give-local-ind-def f f-mono))
+
+\end{code}
+
+We now spell out the correspondence between small 'closed' subsets and
+deflationary points in our sup-lattice. This will allow us to show that
+monotone operators have a least fixed point under certain smallness
+assumpions.
+
+\begin{code}
+
+module _
+        {𝓤 𝓦 𝓥 : Universe}
+        {B : 𝓥  ̇}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        (β : B → ⟨ L ⟩)
+        (h : is-small-basis L β)
+       where
+
+ private
+  _≤_ = order-of L
+  ⋁_ = join-of L
+
+ open Joins _≤_
+ open is-small-basis h
+ open local-inductive-definitions L β h
+
+ module correspondance-from-locally-small-ϕ
+         (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
+         (i : is-local ϕ)
+        where
+
+  is-small-ϕ-closed-subset : (P : 𝓟 {𝓥} B) → 𝓤 ⊔ (𝓥 ⁺)  ̇
```

Comment:
I would write a comment saying that these are really c-closed and phi-closed, or alternatively, rename these to reflect that.